src/pybind/mgr/cephadm/services/mgmt_gateway.py

```diff
@@ -41,36 +68,38 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
         self.mgr.set_module_option_ex('dashboard', 'standby_error_status_code', '503')
         self.mgr.set_module_option_ex('dashboard', 'standby_behaviour', 'error')
 
-    def get_certificates(self, svc_spec: MgmtGatewaySpec, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[str, str, str, str]:
-        self.ssl_certs = SSLCerts()
-        old_cert = self.mgr.cert_key_store.get_cert('mgmt_gw_root_cert')
-        old_key = self.mgr.cert_key_store.get_key('mgmt_gw_root_key')
-        if old_cert and old_key:
-            self.ssl_certs.load_root_credentials(old_cert, old_key)
-        else:
-            self.ssl_certs.generate_root_cert(self.mgr.get_mgr_ip())
-            self.mgr.cert_key_store.save_cert('mgmt_gw_root_cert', self.ssl_certs.get_root_cert())
-            self.mgr.cert_key_store.save_key('mgmt_gw_root_key', self.ssl_certs.get_root_key())
-
+    def get_external_certificates(self, svc_spec: MgmtGatewaySpec, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[str, str]:
+        cert = self.mgr.cert_key_store.get_cert('mgmt_gw_cert')
+        key = self.mgr.cert_key_store.get_key('mgmt_gw_key')
+        if not (cert and key):
+            # not available on store, check if provided on the spec
+            if svc_spec.ssl_certificate and svc_spec.ssl_certificate_key:
```

Comment:
if these can be set in the spec, we should probably update the SpecStore `_save_certs_and_keys` function to store them in the cert key store, rather than treating this one special and pulling it out here.

Reply:
Done (in later commit)